source/MaterialXRender/Handlers/GeometryHandler.cpp

```diff
@@ -1,30 +1,62 @@
+#include <MaterialXGenShader/Util.h>
 #include <MaterialXRender/Handlers/GeometryHandler.h>
 
 namespace MaterialX
 { 
-const std::string GeometryHandler::UNIT_QUAD("unit_quad");
-const std::string GeometryHandler::POSITION_ATTRIBUTE("i_position");
-const std::string GeometryHandler::NORMAL_ATTRIBUTE("i_normal");
-const std::string GeometryHandler::TEXCOORD_ATTRIBUTE("i_texcoord");
-const std::string GeometryHandler::TANGENT_ATTRIBUTE("i_tangent");
-const std::string GeometryHandler::BITANGENT_ATTRIBUTE("i_bitangent");
-const std::string GeometryHandler::COLOR_ATTRIBUTE("i_color");
-
-GeometryHandler::GeometryHandler() :
-    _identifier(UNIT_QUAD)
+void GeometryHandler::addLoader(GeometryLoaderPtr loader)
 {
+    const StringVec& extensions = loader->supportedExtensions();
+    for (auto extension : extensions)
+    {
+        _geometryLoaders.insert(std::pair<std::string, GeometryLoaderPtr>(extension, loader));
+    }
 }
 
-GeometryHandler::~GeometryHandler() 
+bool GeometryHandler::loadGeometry(const std::string& fileName)
 {
-}
+    // Early return if file already loaded
+    if (_fileName == fileName)
```

Comment:
Would it makes sense to keep a set of files instead, an allow multiple calls to loadGeometry to load in several files?

Reply:
I was thinking about this but wanted to leave it simple for now. If needed then can look at an "import". I guess as long as the assumption is that we are not trying to keep more than one "scene" within the handler then I can change this to be cumulative and add in a clear.